enhancements/csvless-bundles.md

```diff
@@ -24,209 +24,488 @@ status: implementable
 
-As a Pipeline Maintainer, Bob wants to add a `registry+v2` bundle to a pre-existing [Index image](https://github.com/operator-framework/olm-book/blob/master/docs/glossary.md#index) which already contains `registry+v1` bundles, so that he can support pipeline targets that switch to `registry+v2`.
+_Ex. Initialize defaults for `registry+v1` in the current empty directory_
 
-By using existing opm commands, Bob will be able to add a `registry+v2` bundle to the pre-existing Index's database and can expect all queries against that database via the [operator-registry API](https://github.com/operator-framework/operator-registry/blob/master/pkg/api/registry.proto) to work. Consequently, Bob can expect any cluster using this Index to correctly surface the respective operator as a PackageManifest.
+```sh
+$ tree .
+.
 
-### Risks and Mitigations
+0 directories, 0 files
+$ opm bundle init --mediatype 'registry+v1' .
+# <logs>
+$ tree .
+.
+├── manifests
+│   └── csv.yaml
+└── metadata
+    ├── annotations.yaml
+    └── dependencies.yaml
 
-One risk is around backporting indexes with mixed mediatypes to older clusters. This needs to be supported to ensure upgrades and downgrades of OpenShift behave as intended. This is outlined further in the Upgrade/Downgrade section below.
+2 directories, 4 files
+```
 
-Another risk is around operatorhub.io - it will need to be adjusted to support the `registry+v2` bundles. This work is not scoped as part of this enhancement and is done by a seperate team - how much effort is involved there is unclear.
+_Ex. Initialize defaults for `registry+v2` in the current empty directory and generate a dockerfile_
+
+```sh
+$ tree .
+.
+
+0 directories, 0 files
+$ opm bundle init --mediatype 'registry+v2' -d .
+# <logs>
+$ tree .
+.
+├── Dockerfile
+├── manifests
+│   └── deployment.yaml
+└── metadata
+    ├── bundle.yaml
+    ├── annotations.yaml
+    └── dependencies.yaml
+
+2 directories, 5 files
+```
 
-Security impacts are minimal because the number of kubernetes objects that OLM supports is still the same - just the delivery mechanism (the bundle format) is changing.
+_Ex. Fail to detect mediatype on empty directory_
 
-## Design Details
+```sh
+$ tree .
+.
 
-### Implementation Details
+0 directories, 0 files
+$ opm bundle init .
+# <log: "unable to infer mediatype when no manifests are present">
+```
+
+_Ex. Infer mediatype from src manifests_
+
+```sh
+$ tree kiali-csvless
+kiali-csvless
+├── crds.yaml
+├── deployment.yaml
+└── rbac.yaml
+$ opm bundle init -o kiali kiali-csvless
+# <logs>
+$ tree kiali
+kiali
+├── manifests
+│   ├── crds.yaml
+│   ├── deployments.yaml
+│   └── rbac.yaml
+└── metadata
+    ├── bundle.yaml
+    ├── annotations.yaml
+    └── dependencies.yaml
+
+2 directories, 6 files
+```
+
+Add a new command to `opm`, `opm bundle build`, which takes as input a directory tree -- described in [File Format](#file-format) -- and outputs an Operator Bundle image.
```

Comment:
Where does the sdk fit in this? Do we want to explicitly subsume bundle building?

Reply:
Really good question.

The reason I added this was to carve out the minimal, dead-simple, UX for handling bundles/indexes and evaluate whether the proposed bundle spec is the reason for any weirdness that may occur (basically reduce the noise of the existing cli and decisions of the other bundle).

We probably don't _need_ to add this, I just wanted to think of the full UX should we subsume that functionality. Plus it's kind of weird that we "support" building indexes with opm but not bundles.

Reply:
Okay, I will end up changing the `alpha` commands in non-backwards compatible ways though -- which is probably fine wrt any definition of support for `alpha`.